include/deal.II/lac/constraint_matrix.h

```diff
@@ -1406,6 +1421,8 @@ ConstraintMatrix::add_entry (const size_type line,
   // exists, since we don't want to enter it twice
   Assert (lines_cache[calculate_line_index(line)] != numbers::invalid_size_type,
           ExcInternalError());
+  Assert (!local_lines.size() || local_lines.is_element(column),
```

Comment:
This check enforces that in a constraint ```x_i =x_j``` j also needs to be locally_relevant, right? My understanding is that this is not required in ```distribute_local_to_global``` because we can write anywhere. I am worried that this change might break some parallel codes (not from hanging node or periodicity constraints of course!).